### DISCUSSION TITLE
Any example of Shutter / blinds control ?

### DISCUSSION THREAD

**Discussion by done7k** [ExperienceLovelace/ha-floorplan#229] [General]
I am looking for an example to control my windows shutters. The issue is there should be three buttons for each shutter motor : UP, STOP, DOWN. and obviously there is no much space available on the plan to place them all for all windows. 

I suspect one of a solution here is to  design some kind of dropdown menu for each window: once window element is active these three control elements appear next to the window to let user to select required action: UP, STOP, DOWN. I am not a pro in CSS but recommended trick to make it happen is to enable children element class like below. 

`
.menu-elements-class { display: none; }

.parent-menu-class:active .menu-elements-class { display: block; }
`

But in SVG those elements are not necessary children, then I have no idea how to change display attribute for their class.

How do you make your shutters manageable on your floorplans?

**Comment by OzGav** [comment]
There are probably a number of ways to attack this. What about touching the icon which causes the action to toggle between UP-STOP-DOWN-STOP and then you could have an arrow which shows the direction the shutter is currently travelling?

**Reply by done7k** [reply]
Unfortunately, my shutter motors (RF433 controlled) do not report current direction and shutter status. But the idea with such function rotating button is nice. the only difficulties I see in a situation when we need to close or open shutter not 100% but half or 2/3 or 3/4 and so on, usually happening at summer. So, if the shutter was closed in half (the button arrow directed to Close-Stop) and we would need to close it "a bit more" the algorithm will be:
- "press" to Open 
- "press" to Stop 
- "press" to Close 
- "press" to Stop at desire position

HA runs some automation to manage the shutters two times a day. I see the shutter control on the floorplan only for manual adjustment at summer time. I do not think we will use it other seasons.

**Reply by OzGav** [reply]
Yes you are correct how it would work in this situation. If you always tend to close it more in summer then you could be clever with your yaml so that the first touch after a period of time is always in the close direction…

**Reply by OzGav** [reply]
Also you could do other things if you always tend to close like tap once to close or stop and double tap to open (or hold)
